### ISSUE TITLE
ACK Controller DBCluster Failure logs not showing

### ISSUE THREAD

**Issue by riddhi2910** [aws-controllers-k8s/community#1447]
**Describe the bug**
I am trying to deploy RDS AURORA DBCluster and GlobalCluster in a single file. While ACK is creating the GlobaclCluster, it fails to find the DB cluster since DBCluster is not ready or failed to create. 

1) The errors in the ACK manager pods only point to why GlobalCluster failed to create which is `ERROR	controller.globalcluster	Reconciler error	{"reconciler group": "rds.services.k8s.aws", "reconciler kind": "GlobalCluster", "name": "servicename", "namespace": "namespace", "error": "DBClusterNotFoundFault: The source cluster could not be found or cannot be accessed: arn:aws:rds:us-east-1:{accountname}:cluster:{dbname}y\n\tstatus code: 404}

2) ACK does not show what is happening to DbCluster if it has some errors or it takes time to provision. but once the globalcluster fails to find the dbcluster, I don't see any other logs other than GlobalCluster failed to create when ACK controller tries to provision the dbscluster

One thing to note here is that we are provisioning DBs in Account A from k8s cluster from Account B we have VPC peering between both. 
**Steps to reproduce**

use combined file:

```yaml
---
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBCluster
metadata:
  labels:
    app: name
  name: name
  namespace: namespace
spec:
  availabilityZones:
  - us-east-1b
  - us-east-1d
  - us-east-1a
  backupRetentionPeriod: 1
  copyTagsToSnapshot: false
  databaseName: rdscanary
  dbClusterIdentifier: region-name
  dbClusterParameterGroupName: name
  dbSubnetGroupName: name
  deletionProtection: false
  enableIAMDatabaseAuthentication: true
  engine: aurora-mysql
  engineMode: provisioned
  engineVersion: 8.0.mysql_aurora.3.02.0
  kmsKeyID: arn:aws:kms:us-east-1:{account}:key/{key_num}
  masterUserPassword:
    key: password
    name: secret name
    namespace: namespace
  masterUsername: root
  port: 3306
  preferredBackupWindow: 03:59-04:29
  preferredMaintenanceWindow: fri:05:59-fri:06:29
  storageEncrypted: true
  tags:
  - key: key
    value: "true"
  vpcSecurityGroupIDs:
  - id1
  - id2
  - id3
---
apiVersion: rds.services.k8s.aws/v1alpha1
kind: GlobalCluster
metadata:
  name: rdscanary
spec:
  globalClusterIdentifier: name
  sourceDBClusterIdentifier: arn:aws:rds:us-east-1:{service}:cluster:us-east-1-service
```

**Expected outcome**
Should be able to provision both dbcluster and globalcluster and logs should be more descriptive 

Environment
AWS

Kubernetes version
1.21+
Using EKS (yes/no), if so version?
YES 1.22
AWS service targeted (S3, RDS, etc.)
RDS Aurora

**Comment by riddhi2910** [comment]
@jkatz any pointers here?

**Comment by jkatz** [comment]
@riddhi2910 We're currently investigating. In your tests, were you able to create a `GlobalCluster` and associated objects one-at-a-time, i.e. adding on each object once the other was available?

**Comment by riddhi2910** [comment]
@jkatz we wait 1800s for Dbcluster to be available before creating `GlobalCluster` once the 1800s done, ACk only shows errors `{"reconciler group": "rds.services.k8s.aws", "reconciler kind": "GlobalCluster", "name": "servicename", "namespace": "namespace", "error": "DBClusterNotFoundFault: The source cluster could not be found or cannot be accessed: arn:aws:rds:us-east-1:{accountname}:cluster:{dbname}y\n\tstatus code: 404}` not really sure if ack is trying to create these objects in bacckground.

**Comment by riddhi2910** [comment]
@jkatz one thing to note here is that we are doing cross-account provisioning, I could create `DBSubnetGroup` but not the `dbcluster` .

**Comment by riddhi2910** [comment]
however other ack controllers (elasticache and dynamodb ) work just fine with cross-account provisioning with VPC peering.

**Comment by riddhi2910** [comment]
@jkatz I checked the destination account, the AWS calls at that account are for `DBSubnetGroup `  and `DBReplicationGroup` and those resources are getting created. I don't see a call to create a `DBCluster `. 

In the ack controller also does not show that it is trying to create the `DBCluster`. Are there any conditions/configs that ack is checking before creating a DBCluster? Can you provide some pointers? 

why there are no logs regarding `DBCluster` ?

**Comment by riddhi2910** [comment]
@jkatz  any pointers? I only see the status on was console now that I am able to create the dbcluster. But, the k8s objects for the dbcluster and globaclcluster do not show any status. Any pointers on why that could be?

**Comment by A-Hilaly** [comment]
Hi @riddhi2910 , can you please provide more information on how did you setup the cross-account? i'll try to reproduce the same in my local environment and see if i can get any answers.

**Comment by riddhi2910** [comment]
Thanks, @A-Hilaly for your reply. We have a `VPC` peering between two accounts. We have a cilium policy to talk to the cross-account security groups. We use deploy rds objects combined via a home-grown tool. I see the calls have been made to the destination accounts and DB is also up and running on the same account. But, I don't see status reported back, and k8s objects of the rds DB do not have `Status`.

**Comment by riddhi2910** [comment]
@A-Hilaly @jkatz So I was able to enable `debug` logs and found this 

`1.663006596896689e+09	DEBUG	ackrt	<< kc.Patch (status)	{"account": "ssss", "role": "", "region": "us-east-1", "kind": "DBCluster", "namespace": "preview", "name": "rdscanary", "is_adopted": false, "generation": 2, "error": "DBCluster.rds.services.k8s.aws \"rdscanary\" is invalid: status.ackResourceMetadata.region: Required value"}` 

the crd for dbcluster is 

> spec:
  conversion:
    strategy: None
  group: rds.services.k8s.aws
  names:
    kind: DBCluster
    listKind: DBClusterList
    plural: dbclusters
    singular: dbcluster
  scope: Namespaced
  versions:
  - name: v1alpha1
    schema:
      openAPIV3Schema:
        description: DBCluster is the Schema for the DBClusters API
        properties:
          apiVersion:
            description: 'APIVersion defines the versioned schema of this representation
              of an object. Servers should convert recognized schemas to the latest
              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
            type: string
          kind:
            description: 'Kind is a string value representing the REST resource this
              object represents. Servers may infer this from the endpoint the client
              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
            type: string
          metadata:
            type: object
          spec:
            description: "DBClusterSpec defines the desired state of DBCluster. \n
              Contains the details of an Amazon Aurora DB cluster. \n This data type
              is used as a response element in the DescribeDBClusters, StopDBCluster,
              and StartDBCluster actions."
            properties:
              availabilityZones:
                description: A list of Availability Zones (AZs) where instances in
                  the DB cluster can be created. For information on Amazon Web Services
                  Regions and Availability Zones, see Choosing the Regions and Availability
                  Zones (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.RegionsAndAvailabilityZones.html)
                  in the Amazon Aurora User Guide.
                items:
                  type: string
                type: array
              backtrackWindow:
                description: "The target backtrack window, in seconds. To disable
                  backtracking, set this value to 0. \n Currently, Backtrack is only
                  supported for Aurora MySQL DB clusters. \n Default: 0 \n Constraints:
                  \n    * If specified, this value must be set to a number from 0
                  to 259,200 (72    hours)."
                format: int64
                type: integer
              backupRetentionPeriod:
                description: "The number of days for which automated backups are retained.
                  \n Default: 1 \n Constraints: \n    * Must be a value from 1 to
                  35"
                format: int64
                type: integer
              characterSetName:
                description: A value that indicates that the DB cluster should be
                  associated with the specified CharacterSet.
                type: string
              copyTagsToSnapshot:
                description: A value that indicates whether to copy all tags from
                  the DB cluster to snapshots of the DB cluster. The default is not
                  to copy them.
                type: boolean
              databaseName:
                description: The name for your database of up to 64 alphanumeric characters.
                  If you do not provide a name, Amazon RDS doesn't create a database
                  in the DB cluster you are creating.
                type: string
              dbClusterIdentifier:
                description: "The DB cluster identifier. This parameter is stored
                  as a lowercase string. \n Constraints: \n    * Must contain from
                  1 to 63 letters, numbers, or hyphens. \n    * First character must
                  be a letter. \n    * Can't end with a hyphen or contain two consecutive
                  hyphens. \n Example: my-cluster1"
                type: string
              dbClusterParameterGroupName:
                description: "The name of the DB cluster parameter group to associate
                  with this DB cluster. If you do not specify a value, then the default
                  DB cluster parameter group for the specified DB engine and version
                  is used. \n Constraints: \n    * If supplied, must match the name
                  of an existing DB cluster parameter    group."
                type: string
              dbSubnetGroupName:
                description: "A DB subnet group to associate with this DB cluster.
                  \n Constraints: Must match the name of an existing DBSubnetGroup.
                  Must not be default. \n Example: mySubnetgroup"
                type: string
              deletionProtection:
                description: A value that indicates whether the DB cluster has deletion
                  protection enabled. The database can't be deleted when deletion
                  protection is enabled. By default, deletion protection is disabled.
                type: boolean
              destinationRegion:
                description: DestinationRegion is used for presigning the request
                  to a given region.
                type: string
              domain:
                description: "The Active Directory directory ID to create the DB cluster
                  in. \n For Amazon Aurora DB clusters, Amazon RDS can use Kerberos
                  Authentication to authenticate users that connect to the DB cluster.
                  For more information, see Kerberos Authentication (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/kerberos-authentication.html)
                  in the Amazon Aurora User Guide."
                type: string
              domainIAMRoleName:
                description: Specify the name of the IAM role to be used when making
                  API calls to the Directory Service.
                type: string
              enableCloudwatchLogsExports:
                description: "The list of log types that need to be enabled for exporting
                  to CloudWatch Logs. The values in the list depend on the DB engine
                  being used. For more information, see Publishing Database Logs to
                  Amazon CloudWatch Logs (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_LogAccess.html#USER_LogAccess.Procedural.UploadtoCloudWatch)
                  in the Amazon Aurora User Guide. \n Aurora MySQL \n Possible values
                  are audit, error, general, and slowquery. \n Aurora PostgreSQL \n
                  Possible value is postgresql."
                items:
                  type: string
                type: array
              enableGlobalWriteForwarding:
                description: "A value that indicates whether to enable this DB cluster
                  to forward write operations to the primary cluster of an Aurora
                  global database (GlobalCluster). By default, write operations are
                  not allowed on Aurora DB clusters that are secondary clusters in
                  an Aurora global database. \n You can set this value only on Aurora
                  DB clusters that are members of an Aurora global database. With
                  this parameter enabled, a secondary cluster can forward writes to
                  the current primary cluster and the resulting changes are replicated
                  back to this cluster. For the primary DB cluster of an Aurora global
                  database, this value is used immediately if the primary is demoted
                  by the FailoverGlobalCluster API operation, but it does nothing
                  until then."
                type: boolean
              enableHTTPEndpoint:
                description: "A value that indicates whether to enable the HTTP endpoint
                  for an Aurora Serverless DB cluster. By default, the HTTP endpoint
                  is disabled. \n When enabled, the HTTP endpoint provides a connectionless
                  web service API for running SQL queries on the Aurora Serverless
                  DB cluster. You can also query your database from inside the RDS
                  console with the query editor. \n For more information, see Using
                  the Data API for Aurora Serverless (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/data-api.html)
                  in the Amazon Aurora User Guide."
                type: boolean
              enableIAMDatabaseAuthentication:
                description: "A value that indicates whether to enable mapping of
                  Amazon Web Services Identity and Access Management (IAM) accounts
                  to database accounts. By default, mapping is disabled. \n For more
                  information, see IAM Database Authentication (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.html)
                  in the Amazon Aurora User Guide."
                type: boolean
              engine:
                description: "The name of the database engine to be used for this
                  DB cluster. \n Valid Values: aurora (for MySQL 5.6-compatible Aurora),
                  aurora-mysql (for MySQL 5.7-compatible Aurora), and aurora-postgresql"
                type: string
              engineMode:
                description: "The DB engine mode of the DB cluster, either provisioned,
                  serverless, parallelquery, global, or multimaster. \n The parallelquery
                  engine mode isn't required for Aurora MySQL version 1.23 and higher
                  1.x versions, and version 2.09 and higher 2.x versions. \n The global
                  engine mode isn't required for Aurora MySQL version 1.22 and higher
                  1.x versions, and global engine mode isn't required for any 2.x
                  versions. \n The multimaster engine mode only applies for DB clusters
                  created with Aurora MySQL version 5.6.10a. \n For Aurora PostgreSQL,
                  the global engine mode isn't required, and both the parallelquery
                  and the multimaster engine modes currently aren't supported. \n
                  Limitations and requirements apply to some DB engine modes. For
                  more information, see the following sections in the Amazon Aurora
                  User Guide: \n    * Limitations of Aurora Serverless (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html#aurora-serverless.limitations)
                  \n    * Limitations of Parallel Query (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-mysql-parallel-query.html#aurora-mysql-parallel-query-limitations)
                  \n    * Limitations of Aurora Global Databases (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-global-database.html#aurora-global-database.limitations)
                  \n    * Limitations of Multi-Master Clusters (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-multi-master.html#aurora-multi-master-limitations)"
                type: string
              engineVersion:
                description: "The version number of the database engine to use. \n
                  To list all of the available engine versions for aurora (for MySQL
                  5.6-compatible Aurora), use the following command: \n aws rds describe-db-engine-versions
                  --engine aurora --query \"DBEngineVersions[].EngineVersion\" \n
                  To list all of the available engine versions for aurora-mysql (for
                  MySQL 5.7-compatible Aurora), use the following command: \n aws
                  rds describe-db-engine-versions --engine aurora-mysql --query \"DBEngineVersions[].EngineVersion\"
                  \n To list all of the available engine versions for aurora-postgresql,
                  use the following command: \n aws rds describe-db-engine-versions
                  --engine aurora-postgresql --query \"DBEngineVersions[].EngineVersion\"
                  \n Aurora MySQL \n Example: 5.6.10a, 5.6.mysql_aurora.1.19.2, 5.7.12,
                  5.7.mysql_aurora.2.04.5 \n Aurora PostgreSQL \n Example: 9.6.3,
                  10.7"
                type: string
              globalClusterIdentifier:
                description: The global cluster ID of an Aurora cluster that becomes
                  the primary cluster in the new global database cluster.
                type: string
              kmsKeyID:
                description: "The Amazon Web Services KMS key identifier for an encrypted
                  DB cluster. \n The Amazon Web Services KMS key identifier is the
                  key ARN, key ID, alias ARN, or alias name for the KMS key. To use
                  a KMS key in a different Amazon Web Services account, specify the
                  key ARN or alias ARN. \n When a KMS key isn't specified in KmsKeyId:
                  \n    * If ReplicationSourceIdentifier identifies an encrypted source,
                  then    Amazon RDS will use the KMS key used to encrypt the source.
                  Otherwise,    Amazon RDS will use your default KMS key. \n    *
                  If the StorageEncrypted parameter is enabled and ReplicationSourceIdentifier
                  \   isn't specified, then Amazon RDS will use your default KMS key.
                  \n There is a default KMS key for your Amazon Web Services account.
                  Your Amazon Web Services account has a different default KMS key
                  for each Amazon Web Services Region. \n If you create a read replica
                  of an encrypted DB cluster in another Amazon Web Services Region,
                  you must set KmsKeyId to a KMS key identifier that is valid in the
                  destination Amazon Web Services Region. This KMS key is used to
                  encrypt the read replica in that Amazon Web Services Region."
                type: string
              kmsKeyRef:
                description: 'AWSResourceReferenceWrapper provides a wrapper around
                  *AWSResourceReference type to provide more user friendly syntax
                  for references using ''from'' field Ex: APIIDRef:   from:     name:
                  my-api'
                properties:
                  from:
                    description: AWSResourceReference provides all the values necessary
                      to reference another k8s resource for finding the identifier(Id/ARN/Name)
                    properties:
                      name:
                        type: string
                    type: object
                type: object
              masterUserPassword:
                description: "The password for the master database user. This password
                  can contain any printable ASCII character except \"/\", \"\"\",
                  or \"@\". \n Constraints: Must contain from 8 to 41 characters."
                properties:
                  key:
                    description: Key is the key within the secret
                    type: string
                  name:
                    description: Name is unique within a namespace to reference a
                      secret resource.
                    type: string
                  namespace:
                    description: Namespace defines the space within which the secret
                      name must be unique.
                    type: string
                required:
                - key
                type: object
              masterUsername:
                description: "The name of the master user for the DB cluster. \n Constraints:
                  \n    * Must be 1 to 16 letters or numbers. \n    * First character
                  must be a letter. \n    * Can't be a reserved word for the chosen
                  database engine."
                type: string
              optionGroupName:
                description: "A value that indicates that the DB cluster should be
                  associated with the specified option group. \n Permanent options
                  can't be removed from an option group. The option group can't be
                  removed from a DB cluster once it is associated with a DB cluster."
                type: string
              port:
                description: "The port number on which the instances in the DB cluster
                  accept connections. \n Default: 3306 if engine is set as aurora
                  or 5432 if set to aurora-postgresql."
                format: int64
                type: integer
              preSignedURL:
                description: "A URL that contains a Signature Version 4 signed request
                  for the CreateDBCluster action to be called in the source Amazon
                  Web Services Region where the DB cluster is replicated from. You
                  only need to specify PreSignedUrl when you are performing cross-region
                  replication from an encrypted DB cluster. \n The pre-signed URL
                  must be a valid request for the CreateDBCluster API action that
                  can be executed in the source Amazon Web Services Region that contains
                  the encrypted DB cluster to be copied. \n The pre-signed URL request
                  must contain the following parameter values: \n    * KmsKeyId -
                  The Amazon Web Services KMS key identifier for the KMS key    to
                  use to encrypt the copy of the DB cluster in the destination Amazon
                  \   Web Services Region. This should refer to the same KMS key for
                  both the    CreateDBCluster action that is called in the destination
                  Amazon Web Services    Region, and the action contained in the pre-signed
                  URL. \n    * DestinationRegion - The name of the Amazon Web Services
                  Region that    Aurora read replica will be created in. \n    * ReplicationSourceIdentifier
                  - The DB cluster identifier for the encrypted    DB cluster to be
                  copied. This identifier must be in the Amazon Resource    Name (ARN)
                  format for the source Amazon Web Services Region. For example,    if
                  you are copying an encrypted DB cluster from the us-west-2 Amazon
                  Web    Services Region, then your ReplicationSourceIdentifier would
                  look like    Example: arn:aws:rds:us-west-2:123456789012:cluster:aurora-cluster1.
                  \n To learn how to generate a Signature Version 4 signed request,
                  see Authenticating Requests: Using Query Parameters (Amazon Web
                  Services Signature Version 4) (https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html)
                  and Signature Version 4 Signing Process (https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html).
                  \n If you are using an Amazon Web Services SDK tool or the CLI,
                  you can specify SourceRegion (or --source-region for the CLI) instead
                  of specifying PreSignedUrl manually. Specifying SourceRegion autogenerates
                  a pre-signed URL that is a valid request for the operation that
                  can be executed in the source Amazon Web Services Region."
                type: string
              preferredBackupWindow:
                description: "The daily time range during which automated backups
                  are created if automated backups are enabled using the BackupRetentionPeriod
                  parameter. \n The default is a 30-minute window selected at random
                  from an 8-hour block of time for each Amazon Web Services Region.
                  To view the time blocks available, see Backup window (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Managing.Backups.html#Aurora.Managing.Backups.BackupWindow)
                  in the Amazon Aurora User Guide. \n Constraints: \n    * Must be
                  in the format hh24:mi-hh24:mi. \n    * Must be in Universal Coordinated
                  Time (UTC). \n    * Must not conflict with the preferred maintenance
                  window. \n    * Must be at least 30 minutes."
                type: string
              preferredMaintenanceWindow:
                description: "The weekly time range during which system maintenance
                  can occur, in Universal Coordinated Time (UTC). \n Format: ddd:hh24:mi-ddd:hh24:mi
                  \n The default is a 30-minute window selected at random from an
                  8-hour block of time for each Amazon Web Services Region, occurring
                  on a random day of the week. To see the time blocks available, see
                  Adjusting the Preferred DB Cluster Maintenance Window (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_UpgradeDBInstance.Maintenance.html#AdjustingTheMaintenanceWindow.Aurora)
                  in the Amazon Aurora User Guide. \n Valid Days: Mon, Tue, Wed, Thu,
                  Fri, Sat, Sun. \n Constraints: Minimum 30-minute window."
                type: string
              replicationSourceIdentifier:
                description: The Amazon Resource Name (ARN) of the source DB instance
                  or DB cluster if this DB cluster is created as a read replica.
                type: string
              scalingConfiguration:
                description: For DB clusters in serverless DB engine mode, the scaling
                  properties of the DB cluster.
                properties:
                  autoPause:
                    type: boolean
                  maxCapacity:
                    format: int64
                    type: integer
                  minCapacity:
                    format: int64
                    type: integer
                  secondsBeforeTimeout:
                    format: int64
                    type: integer
                  secondsUntilAutoPause:
                    format: int64
                    type: integer
                  timeoutAction:
                    type: string
                type: object
              sourceRegion:
                description: SourceRegion is the source region where the resource
                  exists. This is not sent over the wire and is only used for presigning.
                  This value should always have the same region as the source ARN.
                type: string
              storageEncrypted:
                description: A value that indicates whether the DB cluster is encrypted.
                type: boolean
              tags:
                description: Tags to assign to the DB cluster.
                items:
                  description: Metadata assigned to an Amazon RDS resource consisting
                    of a key-value pair.
                  properties:
                    key:
                      type: string
                    value:
                      type: string
                  type: object
                type: array
              vpcSecurityGroupIDs:
                description: A list of EC2 VPC security groups to associate with this
                  DB cluster.
                items:
                  type: string
                type: array
            required:
            - dbClusterIdentifier
            - engine
            type: object
          status:
            description: DBClusterStatus defines the observed state of DBCluster
            properties:
              ackResourceMetadata:
                description: All CRs managed by ACK have a common `Status.ACKResourceMetadata`
                  member that is used to contain resource sync state, account ownership,
                  constructed ARN for the resource
                properties:
                  arn:
                    description: 'ARN is the Amazon Resource Name for the resource.
                      This is a globally-unique identifier and is set only by the
                      ACK service controller once the controller has orchestrated
                      the creation of the resource OR when it has verified that an
                      "adopted" resource (a resource where the ARN annotation was
                      set by the Kubernetes user on the CR) exists and matches the
                      supplied CR''s Spec field values. TODO(vijat@): Find a better
                      strategy for resources that do not have ARN in CreateOutputResponse
                      https://github.com/aws/aws-controllers-k8s/issues/270'
                    type: string
                  ownerAccountID:
                    description: OwnerAccountID is the AWS Account ID of the account
                      that owns the backend AWS service API resource.
                    type: string
                  region:
                    description: Region is the AWS region in which the resource exists
                      or will exist.
                    type: string
                required:
                - ownerAccountID
                - region
                type: object
              activityStreamKMSKeyID:
                description: "The Amazon Web Services KMS key identifier used for
                  encrypting messages in the database activity stream. \n The Amazon
                  Web Services KMS key identifier is the key ARN, key ID, alias ARN,
                  or alias name for the KMS key."
                type: string
              activityStreamKinesisStreamName:
                description: The name of the Amazon Kinesis data stream used for the
                  database activity stream.
                type: string
              activityStreamMode:
                description: The mode of the database activity stream. Database events
                  such as a change or access generate an activity stream event. The
                  database session can handle these events either synchronously or
                  asynchronously.
                type: string
              activityStreamStatus:
                description: The status of the database activity stream.
                type: string
              allocatedStorage:
                description: For all database engines except Amazon Aurora, AllocatedStorage
                  specifies the allocated storage size in gibibytes (GiB). For Aurora,
                  AllocatedStorage always returns 1, because Aurora DB cluster storage
                  size isn't fixed, but instead automatically adjusts as needed.
                format: int64
                type: integer
              associatedRoles:
                description: Provides a list of the Amazon Web Services Identity and
                  Access Management (IAM) roles that are associated with the DB cluster.
                  IAM roles that are associated with a DB cluster grant permission
                  for the DB cluster to access other Amazon Web Services on your behalf.
                items:
                  description: Describes an Amazon Web Services Identity and Access
                    Management (IAM) role that is associated with a DB cluster.
                  properties:
                    featureName:
                      type: string
                    roleARN:
                      type: string
                    status:
                      type: string
                  type: object
                type: array
              automaticRestartTime:
                description: The time when a stopped DB cluster is restarted automatically.
                format: date-time
                type: string
              backtrackConsumedChangeRecords:
                description: The number of change records stored for Backtrack.
                format: int64
                type: integer
              capacity:
                description: "The current capacity of an Aurora Serverless DB cluster.
                  The capacity is 0 (zero) when the cluster is paused. \n For more
                  information about Aurora Serverless, see Using Amazon Aurora Serverless
                  (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html)
                  in the Amazon Aurora User Guide."
                format: int64
                type: integer
              cloneGroupID:
                description: Identifies the clone group to which the DB cluster is
                  associated.
                type: string
              clusterCreateTime:
                description: Specifies the time when the DB cluster was created, in
                  Universal Coordinated Time (UTC).
                format: date-time
                type: string
              conditions:
                description: All CRS managed by ACK have a common `Status.Conditions`
                  member that contains a collection of `ackv1alpha1.Condition` objects
                  that describe the various terminal states of the CR and its backend
                  AWS service API resource
                items:
                  description: Condition is the common struct used by all CRDs managed
                    by ACK service controllers to indicate terminal states  of the
                    CR and its backend AWS service API resource
                  properties:
                    lastTransitionTime:
                      description: Last time the condition transitioned from one status
                        to another.
                      format: date-time
                      type: string
                    message:
                      description: A human readable message indicating details about
                        the transition.
                      type: string
                    reason:
                      description: The reason for the condition's last transition.
                      type: string
                    status:
                      description: Status of the condition, one of True, False, Unknown.
                      type: string
                    type:
                      description: Type is the type of the Condition
                      type: string
                  required:
                  - status
                  - type
                  type: object
                type: array
              crossAccountClone:
                description: Specifies whether the DB cluster is a clone of a DB cluster
                  owned by a different Amazon Web Services account.
                type: boolean
              customEndpoints:
                description: Identifies all custom endpoints associated with the cluster.
                items:
                  type: string
                type: array
              dbClusterMembers:
                description: Provides the list of instances that make up the DB cluster.
                items:
                  description: Contains information about an instance that is part
                    of a DB cluster.
                  properties:
                    dbClusterParameterGroupStatus:
                      type: string
                    dbInstanceIdentifier:
                      type: string
                    isClusterWriter:
                      type: boolean
                    promotionTier:
                      format: int64
                      type: integer
                  type: object
                type: array
              dbClusterOptionGroupMemberships:
                description: Provides the list of option group memberships for this
                  DB cluster.
                items:
                  description: Contains status information for a DB cluster option
                    group.
                  properties:
                    dbClusterOptionGroupName:
                      type: string
                    status:
                      type: string
                  type: object
                type: array
              dbClusterParameterGroup:
                description: Specifies the name of the DB cluster parameter group
                  for the DB cluster.
                type: string
              dbClusterResourceID:
                description: The Amazon Web Services Region-unique, immutable identifier
                  for the DB cluster. This identifier is found in Amazon Web Services
                  CloudTrail log entries whenever the KMS key for the DB cluster is
                  accessed.
                type: string
              dbSubnetGroup:
                description: Specifies information on the subnet group associated
                  with the DB cluster, including the name, description, and subnets
                  in the subnet group.
                type: string
              domainMemberships:
                description: The Active Directory Domain membership records associated
                  with the DB cluster.
                items:
                  description: An Active Directory Domain membership record associated
                    with the DB instance or cluster.
                  properties:
                    domain:
                      type: string
                    fQDN:
                      type: string
                    iamRoleName:
                      type: string
                    status:
                      type: string
                  type: object
                type: array
              earliestBacktrackTime:
                description: The earliest time to which a DB cluster can be backtracked.
                format: date-time
                type: string
              earliestRestorableTime:
                description: The earliest time to which a database can be restored
                  with point-in-time restore.
                format: date-time
                type: string
              enabledCloudwatchLogsExports:
                description: "A list of log types that this DB cluster is configured
                  to export to CloudWatch Logs. \n Log types vary by DB engine. For
                  information about the log types for each DB engine, see Amazon RDS
                  Database Log Files (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_LogAccess.html)
                  in the Amazon Aurora User Guide."
                items:
                  type: string
                type: array
              endpoint:
                description: Specifies the connection endpoint for the primary instance
                  of the DB cluster.
                type: string
              globalWriteForwardingRequested:
                description: Specifies whether you have requested to enable write
                  forwarding for a secondary cluster in an Aurora global database.
                  Because write forwarding takes time to enable, check the value of
                  GlobalWriteForwardingStatus to confirm that the request has completed
                  before using the write forwarding feature for this cluster.
                type: boolean
              globalWriteForwardingStatus:
                description: Specifies whether a secondary cluster in an Aurora global
                  database has write forwarding enabled, not enabled, or is in the
                  process of enabling it.
                type: string
              hostedZoneID:
                description: Specifies the ID that Amazon Route 53 assigns when you
                  create a hosted zone.
                type: string
              httpEndpointEnabled:
                description: "A value that indicates whether the HTTP endpoint for
                  an Aurora Serverless DB cluster is enabled. \n When enabled, the
                  HTTP endpoint provides a connectionless web service API for running
                  SQL queries on the Aurora Serverless DB cluster. You can also query
                  your database from inside the RDS console with the query editor.
                  \n For more information, see Using the Data API for Aurora Serverless
                  (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/data-api.html)
                  in the Amazon Aurora User Guide."
                type: boolean
              iamDatabaseAuthenticationEnabled:
                description: A value that indicates whether the mapping of Amazon
                  Web Services Identity and Access Management (IAM) accounts to database
                  accounts is enabled.
                type: boolean
              latestRestorableTime:
                description: Specifies the latest time to which a database can be
                  restored with point-in-time restore.
                format: date-time
                type: string
              multiAZ:
                description: Specifies whether the DB cluster has instances in multiple
                  Availability Zones.
                type: boolean
              pendingModifiedValues:
                description: A value that specifies that changes to the DB cluster
                  are pending. This element is only included when changes are pending.
                  Specific changes are identified by subelements.
                properties:
                  dbClusterIdentifier:
                    type: string
                  engineVersion:
                    type: string
                  iamDatabaseAuthenticationEnabled:
                    type: boolean
                  masterUserPassword:
                    type: string
                  pendingCloudwatchLogsExports:
                    description: A list of the log types whose configuration is still
                      pending. In other words, these log types are in the process
                      of being activated or deactivated.
                    properties:
                      logTypesToDisable:
                        items:
                          type: string
                        type: array
                      logTypesToEnable:
                        items:
                          type: string
                        type: array
                    type: object
                type: object
              percentProgress:
                description: Specifies the progress of the operation as a percentage.
                type: string
              readReplicaIdentifiers:
                description: Contains one or more identifiers of the read replicas
                  associated with this DB cluster.
                items:
                  type: string
                type: array
              readerEndpoint:
                description: "The reader endpoint for the DB cluster. The reader endpoint
                  for a DB cluster load-balances connections across the Aurora Replicas
                  that are available in a DB cluster. As clients request new connections
                  to the reader endpoint, Aurora distributes the connection requests
                  among the Aurora Replicas in the DB cluster. This functionality
                  can help balance your read workload across multiple Aurora Replicas
                  in your DB cluster. \n If a failover occurs, and the Aurora Replica
                  that you are connected to is promoted to be the primary instance,
                  your connection is dropped. To continue sending your read workload
                  to other Aurora Replicas in the cluster, you can then reconnect
                  to the reader endpoint."
                type: string
              scalingConfigurationInfo:
                description: "Shows the scaling configuration for an Aurora DB cluster
                  in serverless DB engine mode. \n For more information, see Using
                  Amazon Aurora Serverless (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html)
                  in the Amazon Aurora User Guide."
                properties:
                  autoPause:
                    type: boolean
                  maxCapacity:
                    format: int64
                    type: integer
                  minCapacity:
                    format: int64
                    type: integer
                  secondsBeforeTimeout:
                    format: int64
                    type: integer
                  secondsUntilAutoPause:
                    format: int64
                    type: integer
                  timeoutAction:
                    type: string
                type: object
              status:
                description: Specifies the current state of this DB cluster.
                type: string
              tagList:
                items:
                  description: Metadata assigned to an Amazon RDS resource consisting
                    of a key-value pair.
                  properties:
                    key:
                      type: string
                    value:
                      type: string
                  type: object
                type: array
              vpcSecurityGroups:
                description: Provides a list of VPC security groups that the DB cluster
                  belongs to.
                items:
                  description: This data type is used as a response element for queries
                    on VPC security group membership.
                  properties:
                    status:
                      type: string
                    vpcSecurityGroupID:
                      type: string
                  type: object
                type: array
            type: object
        type: object
    served: true
    storage: true
    subresources:
      status: {}
status:
  acceptedNames:
    kind: DBCluster
    listKind: DBClusterList
    plural: dbclusters
    singular: dbcluster
  conditions:
  - lastTransitionTime: "2022-05-17T22:30:59Z"
    message: no conflicts found
    reason: NoConflicts
    status: "True"
    type: NamesAccepted
  - lastTransitionTime: "2022-05-17T22:30:59Z"
    message: the initial names have been accepted
    reason: InitialNamesAccepted
    status: "True"
    type: Established
  storedVersions:
  - v1alpha1

**Comment by A-Hilaly** [comment]
Hey @riddhi2910, 

So the way we set regions in ACK land is by setting annotations, similar to this config https://github.com/aws-controllers-k8s/lambda-controller/blob/main/test/e2e/resources/alias.yaml#L6.

Are also setting CARM using namespaces and the "carm" configmap? as mentioned here https://aws-controllers-k8s.github.io/community/docs/user-docs/cross-account-resource-management/

Also happy to hop on a Slack call and discuss/debug this with you, my name on slack is Amine Hilaly.

**Comment by A-Hilaly** [comment]
This issue was solved by using rds-controller `v0.1.0`
/close
